client-c/HttpHandler.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "HttpHandler.h"
+
+#include <proxygen/httpserver/RequestHandler.h>
+#include <proxygen/httpserver/ResponseBuilder.h>
+
+#include "HttpStats.h"
+
+#include <iostream>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <cryptopp/sha.h>
+#include <cryptopp/filters.h>
+#include <cryptopp/hex.h>
+
+using namespace proxygen;
+
+namespace HttpService {
+
+HttpHandler::HttpHandler(HttpStats* stats, int k_socket): stats_(stats),
+                                                          k_socket_(k_socket),
+                                                          bracket_counter_(0),
+                                                          request_method_(folly::none) {
+}
+
+void HttpHandler::onRequest(std::unique_ptr<HTTPMessage> headers) noexcept {
+  stats_->recordRequest();
+  request_method_ = headers -> getMethod();
+}
+
+void HttpHandler::onBody(std::unique_ptr<folly::IOBuf> r_body) noexcept {
+  if (HTTPMethod::POST == request_method_) {
+    std::string body = r_body -> moveToFbString().toStdString();
+    std::string message;
+    char buffer[2] = {0};
```

Comment:
having such a small buffer seems like it would be slow; we should read at least a 1-4k bytes at a time; I understand it makes `doneReading` easier to implement, but it shouldn't be that hard to count characters in a char array.

Alternatively, if there was a way to prepend the length of the message in the K client such that we know how much to read in advance, that might help with the code here. Maybe @ehildenb can comment on the viability of this suggestion.